lib/game.rb

```diff
@@ -0,0 +1,96 @@
+require 'pry'
+
+class Game
+
+  attr_accessor :player_1, :player_2, :board
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [2,4,6]
+    ]
+
+    def initialize(player_1 = Human.new("X"), player_2 = Human.new("O"), board = Board.new)
+      @player_1 = player_1
+      @player_2 = player_2
+      @board = board
+      @winner = nil
+    end
+
+    def current_player
+      turn_counter = board.cells.count{|a| a != " "}
+      if turn_counter % 2 == 0
+        return @player_1
+      else
+        return @player_2 
+      end
+    end
+
+    def over?
+      draw? || won?
+    end
+
+    def won?
+      cb = board.cells
+      WIN_COMBINATIONS.each do |a|
+        if (cb[a[0]] != " ") && (cb[a[0]] == cb[a[1]] && cb[a[0]] == cb[a[2]])
+          @winner = cb[a[0]]
+          return true
+        end
+      end
+      return false
+    end
+
```

Comment:
This is good, I'd consider this:

```ruby
def won?
  WIN_COMBINATIONS.each do |a|
    @winner = board.cells[a[0]] if (board.cells[a[0]] != " ") && 
                                   (board.cells[a[0]] == board.cells[a[1]] && 
                                    board.cells[a[0]] == board.cells[a[2]])
  end

  return @winner
end
```

1. Don't create local variables, use `board.cells`. One reason is you could simplify your `Board` instances by building something like `Board#position` or even `Board#[]` that creates a short hand accessor for the value of the board's cell at a position.

```ruby
class Board
  def [](index)
    self.cells[index]
  end

  def position(index)
    self.cells[index]
  end
end
```

Then you could write:

```ruby
def won?
  WIN_COMBINATIONS.each do |a|
    @winner = board[a[0]] if (board[a[0]] != " ") && 
                                   (board[a[0]] == board[a[1]] && 
                                    board[a[0]] == board[a[2]])
  end

  return @winner
end
```

Pretty cool that you can implement your own `Object#[]` and make your own objects have an array like interface, right? :heart: ruby.

And then instead of returning true or false, I'm returning `@winner`. I, and I alone, like a `return` keyword for the last line of the method when I'm returning a somewhat unrelated value like an instance variable or a local variable. But the idea is that either `@winner` was set correctly or it wasn't set at all in which case the method returns nil.